File: golang/cbnu_2021_2_online_contest/7/7.go
package main

import "fmt"

func main() {
	var tc, t int
	result := [...]string{"NA", "MMYY", "YYMM", "AMBIGOUS"}
	fmt.Scanln(&t)
	for tc = 1; tc <= t; tc++ {
		var c [4]rune
		result_num := 0
		fmt.Scanf("%s", c)
		frontNum := (c[0]-'0')*10 + (c[1] - '0')
		backNum := (c[2]-'0')*10 + (c[3] - '0')
		if frontNum < 13 && backNum > 0 {
			result_num += 1
		}
		if backNum < 13 && frontNum > 0 {
			result_num += 2
		}
		fmt.Printf("#%d %s\n", tc, result[result_num])
	}
}

File: golang/cbnu_2021_2_online_contest/7/7.cpp
#include <iostream>
#include <cstdio>
#include <string>
using namespace std;

int main(){
    int tc;
    string result[]={"NA","MMYY","YYMM","AMBIGUOUS"};
    int t;
    scanf("%d",&t);

    for(tc = 1; tc<= t; ++tc) {
    char s[4];
    int r_n=0;
    scanf("%s",s);
    int fN=10*(s[0]-'0') +(s[1]-'0');
    int bN=10*(s[2]-'0') +(s[3]-'0');

    if (fN<13 && bN>0) r_n+=1;
    if (bN<13 && fN>0) r_n+=2;

    printf("#%d %s\n",tc,result[r_n].c_str());
    }
}